client/ayon_core/scripts/otio_burnin.py

```diff
@@ -482,10 +483,19 @@ def render(self, output, args=None, overwrite=False, **kwargs):
         )
         print("Launching command: {}".format(command))
 
+        use_shell = True
+        try:
+            test_proc = subprocess.Popen(
+                f"{FFMPEG_EXE_COMMAND} --help", shell=True
+            )
+            test_proc.wait()
+        except BaseException:
+            use_shell = False
+
         kwargs = {
             "stdout": subprocess.PIPE,
             "stderr": subprocess.PIPE,
-            "shell": True,
+            "shell": use_shell,
```

Comment:
Don't these subprocess calls technically mean they may show up a command prompt dialog on Windows? Shouldn't these have that "hide console window" special flag?

Also, it's really annoying we have to do this because these subprocess calls startup always take longer than I'd like. I mean it's better than failing.

Maybe we could cache the "should we run in shell" result for the running process? (Does that even make sense since this would be a script that would maybe already itself be running in a subprocess maybe?)

Reply:
> Also, it's really annoying we have to do this because these subprocess calls startup always take longer than I'd like. I mean it's better than failing.
>
> Maybe we could cache the "should we run in shell" result for the running process? (Does that even make sense since this would be a script that would maybe already itself be running in a subprocess maybe?)

This is called only once during the process lifetime (single extract burning process).

Ideal fix is to rewrite the logic so output is list of arguments, in that case we can get rid of `shell=True`.

Reply:
Yup - thanks for confirming.